SeleniumAutomationGenerator/ComponentFactorySteps/IComponentFactoryCreatorStep.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections.Generic;
+using Core;
+using Core.Models;
+
+namespace SeleniumAutomationGenerator.ComponentFactorySteps
+{
+    public interface IComponentFactoryCreatorStep : IComponentFactoryStep
+    {
+        IEnumerable<ComponentGeneratorOutput> InvokeStep(AutoElementData rootElement, IComponentFileCreator parent, IComponentFactoryCreatorStep next = null);
```

Comment:
in some of the cases `parent` is null but based on the signature it's not it clear that it could be null.
either make all null-able parameters be  null-able or add overloads with less parameters.

A better solution might be to add to override the method in `ComponentFactoryCreatorStep` and add virtual methods with less parameters